logstash-core/api/lib/app.rb

```diff
@@ -4,16 +4,26 @@
 require "app_helpers"
 require "app/service"
 require "app/command_factory"
+require "logstash/util/loggable"
 
 module LogStash::Api
   class BaseApp < ::Sinatra::Application
 
     attr_reader :factory
 
+    ::Cabin::Channel.class_eval { alias :write :publish }
```

Comment:
What @jordansissel is suggesting is to create a module adding the compatibility layer required by Sinatra instead of using class_eval.

Something like 
```ruby
::Cabin::Channel.extend(Cabin::SinatraCompatibility)
```

The module can live inside logstash in the util directory, this is a bit cleaner and we don't need to push this change upstream.